eng/build.ps1

```diff
@@ -374,6 +376,10 @@ function TestUsingRunTests() {
     $env:ROSLYN_TEST_IOPERATION = "true"
   }
 
+  if ($testEmitDiagnostics) {
+    $env:ROSLYN_TEST_EMITDIAGNOSTICS = "true"
+  }
+
   if ($testUsedAssemblies) {
```

Comment:
@AlekseyTs Should we enable `testUsedAssemblies` in a CI leg? Seems like it's unused at the moment.

Reply:
> Should we enable testUsedAssemblies in a CI leg? Seems like it's unused at the moment.

I think it would be good to add a separate CI leg similar to the one used for IOperation testing.

Reply:
Okay. Let's keep that separate (there's a few questions, such as choosing a leg, how to deal with potential failures that might have snuck in). 
Filed tracking [issue](https://github.com/dotnet/roslyn/issues/60086).

Towards that goal, this PR fixes an issue with TestRunner that makes sure we copy environment variables to Helix machines where tests end up running.

Reply:
I think we should consider whether to group this set of "additional validation" flags into a single leg or to have separate legs. Recently we are trying to be sensitive about number of test legs introduced because:

1. Each time we additionally run the tests increases the rate of spurious failures.
2. Cost of provisioning the additional machines.

See #57313.